lib/ecto/migrator.ex

```diff
@@ -105,13 +120,27 @@ defmodule Ecto.Migrator do
       attempt(repo, module, :forward, :down, :down, opts)
         || attempt(repo, module, :backward, :change, :down, opts)
         || raise Ecto.MigrationError, "#{inspect module} does not implement a `down/0` or `change/0` function"
-      verbose_schema_migration repo, "update schema migrations", fn ->
-        SchemaMigration.down(repo, version, opts[:prefix])
-      end
+    end
+
+    verbose_schema_migration repo, "update schema migrations", fn ->
+      SchemaMigration.down(repo, version, opts[:prefix])
     end
   end
 
   defp run_maybe_in_transaction(repo, module, fun) do
+    Task.async(fn ->
+      do_run_maybe_in_transaction(repo, module, fun)
+    end)
+    |> Task.await(:infinity)
+    |> case do
+      {:error, error, stacktrace} ->
```

Comment:
A task would never return this. Instead it would cause the linked process to fail.